### ISSUE TITLE
CPU Temperature Indicator

### ISSUE THREAD

**Issue by Padre2** [linuxmint/cinnamon-spices-applets#2706]
UUID: temperature@fevimu

Applet version/Build date - Updated on 2019-11-27 03:57:45

Cinnamon version 4.4

Distribution - Arch

Graphics hardware AMD

After update is working, but floods the .Xsession-errors file with warnings:
_Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested)_ 
This stops if removed and repeats when loaded.

**Comment by claudiux** [comment]
@Padre2 Next time mention me, please: @claudiux. I will try to install Arch on a real machine, not a virtual one, to debug this.

**Comment by Padre2** [comment]
@claudiux  After further checking I can confirm that this CPU temperature applet is displaying the GPU temperature in the panel. This is the last sensor (2nd temp1) in the following list:
sensor.log
S,lmsensor it8720-isa-0228 temp1,101
S,lmsensor it8720-isa-0228 temp2,101
S,lmsensor it8720-isa-0228 temp3,101
S,lmsensor radeon-pci-0100 temp1,101

This is further confirmed by checking on same machine running Manjaro OS with Cinnamon 4.2.4, which only displays the (same) GPU temperature.

Also, if I load the  gputemperature@silentage.com applet, it displays correctly (only) the same GPU temperature values on both Arch and Manjaro.  I do not leave this applet running as it displays a  _not compatible warning_  on loading and crashes Cinnamon from time to time.  I posted the issue several days ago, but have had no response from @silentage.com. My priority would be to get the GPU temperature running rather than the CPU, because I never overclock and have not had any CPU overheating issues.

**Comment by Padre2** [comment]
@claudiux  Just updated this app.
The temperature showing from temperature@fevimu in the panel is definitely from the GPU (AMD Radeon) and not the CPU.
The it8720-isa-0228 temp1 is the CPU, temp2 is the mobo and temp 3 is unknown.  
In searching this, many give temp3 ~82. Never changes and nobody seems to know what it is.

**Comment by claudiux** [comment]
@Padre2 I'm searching how to solve these issues...

**Comment by Padre2** [comment]
@claudiux Thank you, I shall be interested to learn more about these issues.

Running sensors-detect: 
Driver `it87':
  * ISA bus, address 0x228
    Chip `ITE IT8720F Super IO Sensors'

  Driver `i2c-piix4' for device 0000:00:14.0:
  ATI Technologies Inc SB600/SB700/SB800 SMBus

In the meantime, I installed xsensors, which displays the values correctly for  GPU, CPU and MOBO.
![it872-_sensor_2019-12-10](https://user-images.githubusercontent.com/38849451/70496222-6e905b80-1b41-11ea-9e6e-f37931706737.png)
It refers to the GPU sensor only as Radeon.

**Comment by claudiux** [comment]
@Padre2 Can you also post the second page, please?

**Comment by Padre2** [comment]
> Can you also post the second page, please?
@claudiux Sorry for delay, first major issue with 4 months with my Arch compilation:  no sound after updates - spent hours fixing and tweaking ALSA and PulseAudio - learned a lot.  :-)
I presume you mean the Radeon page. Not much to see, please find below.
![Radeon_sensor_2019-12-11](https://user-images.githubusercontent.com/38849451/70622154-ddf37180-1c4d-11ea-8361-cea49851144c.png)

**Comment by claudiux** [comment]
@Padre2 Thank you. I think I have found where the problem comes from. It only remains for me to solve it!

**Comment by Padre2** [comment]
@claudiux  I tried changing `hwmon0` to `hwmon1` as in the following, starting line 273 in the applet.js and was able to get it to show the CPU temperature in the panel.  Is this enough to fix this issue?

`_findTemperatureFromFiles: function() {
    let info = {};
    let tempFiles = [
      // hwmon for new 2.6.39, 3.x linux kernels
      '/sys/class/hwmon/hwmon1/temp1_input',
      '/sys/devices/platform/coretemp.0/temp1_input',
      '/sys/bus/acpi/devices/LNXTHERM:00/thermal_zone/temp',
      '/sys/devices/virtual/thermal/thermal_zone0/temp',
      '/sys/bus/acpi/drivers/ATK0110/ATK0110:00/hwmon/hwmon1z/temp1_input',
      // old kernels with proc fs
      '/proc/acpi/thermal_zone/THM0/temperature',
      '/proc/acpi/thermal_zone/THRM/temperature',
      '/proc/acpi/thermal_zone/THR0/temperature',
      '/proc/acpi/thermal_zone/TZ0/temperature',
      // Debian Sid/Experimental on AMD-64
      '/sys/class/hwmon/hwmon1/device/temp1_input'
    ];
    for (let i = 0; i < tempFiles.length; i++) {
      if (GLib.file_test(tempFiles[i], 1 << 4)) {
        let temperature = GLib.file_get_contents(tempFiles[i]);
        if (temperature[0]) {
          info.temp = parseInt(temperature[1]) / 1000;
          break;
        }
      }
    }`

Btw, running this applet with the above, floods .Xsession-errors file with
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested)

Regarding the critical temperature, where can this be set? It should be possible to configure it.  While there may be some Intel mobile CPU's that can go to 100 °C, the recommended high operating temperature for my AMD CPU is 70 °C.

**Comment by Padre2** [comment]
@claudiux  I tried changing `hwmon0` to `hwmon1` as in the following, starting line 273 in the applet.js and was able to get it to show the CPU temperature in the panel.  Is this enough to fix this issue?

`_findTemperatureFromFiles: function() {
    let info = {};
    let tempFiles = [
      // hwmon for new 2.6.39, 3.x linux kernels
      '/sys/class/hwmon/hwmon1/temp1_input',
      '/sys/devices/platform/coretemp.0/temp1_input',
      '/sys/bus/acpi/devices/LNXTHERM:00/thermal_zone/temp',
      '/sys/devices/virtual/thermal/thermal_zone0/temp',
      '/sys/bus/acpi/drivers/ATK0110/ATK0110:00/hwmon/hwmon1z/temp1_input',
      // old kernels with proc fs
      '/proc/acpi/thermal_zone/THM0/temperature',
      '/proc/acpi/thermal_zone/THRM/temperature',
      '/proc/acpi/thermal_zone/THR0/temperature',
      '/proc/acpi/thermal_zone/TZ0/temperature',
      // Debian Sid/Experimental on AMD-64
      '/sys/class/hwmon/hwmon1/device/temp1_input'
    ];
    for (let i = 0; i < tempFiles.length; i++) {
      if (GLib.file_test(tempFiles[i], 1 << 4)) {
        let temperature = GLib.file_get_contents(tempFiles[i]);
        if (temperature[0]) {
          info.temp = parseInt(temperature[1]) / 1000;
          break;
        }
      }
    }`

Btw, running this applet with the above, floods .Xsession-errors file with
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested)

Regarding the critical temperature, where can this be set? It should be possible to configure it.  While there may be some Intel mobile CPU's that can go to 120 °C, the recommended high operating temperature for my AMD CPU is 70 °C.

**Comment by claudiux** [comment]
@Padre2 

I will add some files in this list.

This Cinnamon warning only appears on Arch. I don't know why. I'm searching...

**Comment by claudiux** [comment]
@Padre2 Can you post (using triple back-quote) the result of this command, please?
`find /sys -type f -name "temp*"`

**Comment by Padre2** [comment]
@claudiux   Run with sudo:
```/sys/devices/platform/it87.552/hwmon/hwmon1/temp2_offset
/sys/devices/platform/it87.552/hwmon/hwmon1/temp2_min
/sys/devices/platform/it87.552/hwmon/hwmon1/temp2_type
/sys/devices/platform/it87.552/hwmon/hwmon1/temp3_offset
/sys/devices/platform/it87.552/hwmon/hwmon1/temp3_alarm
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_beep
/sys/devices/platform/it87.552/hwmon/hwmon1/temp2_max
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_type
/sys/devices/platform/it87.552/hwmon/hwmon1/temp3_input
/sys/devices/platform/it87.552/hwmon/hwmon1/temp3_min
/sys/devices/platform/it87.552/hwmon/hwmon1/temp3_max
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_alarm
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_input
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_min
/sys/devices/platform/it87.552/hwmon/hwmon1/temp3_beep
/sys/devices/platform/it87.552/hwmon/hwmon1/temp2_alarm
/sys/devices/platform/it87.552/hwmon/hwmon1/temp3_type
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_max
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_offset
/sys/devices/platform/it87.552/hwmon/hwmon1/temp2_input
/sys/devices/platform/it87.552/hwmon/hwmon1/temp2_beep
/sys/devices/pci0000:00/0000:00:02.0/0000:01:00.0/hwmon/hwmon0/temp1_crit
/sys/devices/pci0000:00/0000:00:02.0/0000:01:00.0/hwmon/hwmon0/temp1_crit_hyst
/sys/devices/pci0000:00/0000:00:02.0/0000:01:00.0/hwmon/hwmon0/temp1_input

Thank you for the triple backquote - had to search it.  :-)

**Comment by Padre2** [comment]
> This Cinnamon warning only appears on Arch. I don't know why. I'm searching...

This often happens only once for some applets and is not really a problem, but I object when it floods the page.  I am trying to discover why and how this is generated.

**Comment by claudiux** [comment]
@Padre2 On my computer, the `sudo find /sys -type f -name "temp*" | sort` command displays:
```
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp1_crit
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp1_crit_alarm
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp1_input
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp1_label
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp1_max
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp2_crit
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp2_crit_alarm
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp2_input
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp2_label
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp2_max
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp3_crit
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp3_crit_alarm
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp3_input
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp3_label
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp3_max
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp4_crit
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp4_crit_alarm
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp4_input
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp4_label
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp4_max
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp5_crit
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp5_crit_alarm
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp5_input
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp5_label
/sys/devices/platform/coretemp.0/hwmon/hwmon1/temp5_max
/sys/devices/virtual/hwmon/hwmon0/temp1_crit
/sys/devices/virtual/hwmon/hwmon0/temp1_input
/sys/devices/virtual/hwmon/hwmon0/temp2_crit
/sys/devices/virtual/hwmon/hwmon0/temp2_input
/sys/devices/virtual/thermal/thermal_zone0/temp
/sys/devices/virtual/thermal/thermal_zone1/temp
/sys/devices/virtual/thermal/thermal_zone2/temp
```
As you can see, some filenames are completely different.

Can you publish the contents of the following files, please? I will try to automate the detection of "useful files", and I want to be sure of their role.
```
/sys/devices/pci0000:00/0000:00:02.0/0000:01:00.0/hwmon/hwmon0/temp1_crit
/sys/devices/pci0000:00/0000:00:02.0/0000:01:00.0/hwmon/hwmon0/temp1_crit_hyst
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_alarm
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_beep
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_max
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_min
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_offset
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_type
```

**Comment by claudiux** [comment]
@Padre2 Did you try to use the command `sudo sensors-detect`?

**Comment by Padre2** [comment]
@claudiux   `running-sensors` detect with `sudo` does not seem to produce any more information than I pasted above.

**Comment by Padre2** [comment]
@claudiux   `running-sensors-detect` with `sudo` does not seem to produce any more information than I pasted above.

**Comment by claudiux** [comment]
@Padre2 Can you try this command, please? `sudo modprobe coretemp`.

If successful: 

- Replace `hwmon1` by `hwmon0` in the tempFiles list.

- Restart this applet (or Cinnamon) and open its menu. Can you see any change?


(I'm trying to rewrite this applet to autodetect the right temperature sensors. What a fog!)

**Comment by Padre2** [comment]
@claudiux   `sudo modprobe coretemp`
`modprobe: ERROR: could not insert 'coretemp': No such device`
You did not comment on the changes I posted above, which do change the display to the CPU temperature on the panel.

**Comment by Padre2** [comment]
@claudiux   > What a fog!)
Have you read this?
[https://www.fnal.gov/projects/ckm/ckm_cluster/sysadmin/lm-sensors/sensors.conf.ckmxx]
Apologies, if it only adds to the fog.

**Comment by claudiux** [comment]
@Padre2 Yes, you can add `'/sys/class/hwmon/hwmon1/temp1_input',` as first element of tempFiles and `'/sys/class/hwmon/hwmon1/temp1_crit',` as first element of critFiles.

Thank you for the link. Each system has its own file `/etc/sensors.conf` or `/etc/sensors3.conf`. That will help me to determine the CPU temperature sensor.

I don't understand why, on your system, this applet does not use the data returned by the `sensors` command. Can you copy here the entire return of this command, please?

**Comment by claudiux** [comment]
@Padre2 I just made the PR #2755. Can you test the new `applet.js`, please?

**Comment by Padre2** [comment]
@claudiux   Downloaded the applet.js patch, added the original first 43 lines and replaced the original with the new file. I presume this is the correct procedure.  Loads OK and looks good! The applet is showing the correct CPU temperature in the panel.   Left click shows other temperatures as before.  Below are copies of relevant parts of the .Xsession-errors file without the applet loaded and after the applet loaded.
```Cjs-Message: 13:16:33.994: JS LOG: [LookingGlass/info] Loaded applet network@cinnamon.org in 146 ms
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cjs-Message: 13:16:34.306: JS LOG: [LookingGlass/info] Loaded applet SpicesUpdate@claudiux in 312 ms
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cjs-Message: 13:16:35.242: JS LOG: [LookingGlass/info] Loaded applet menu@cinnamon.org in 936 ms
Cjs-Message: 13:16:35.243: JS LOG: [LookingGlass/info] AppletManager started in 2886 ms
Cjs-Message: 13:16:35.248: JS LOG: [LookingGlass/info] Cinnamon took 3342 ms to start
Cjs-Message: 13:16:35.346: JS LOG: Unknown network device type, is 14
Cjs-Message: 13:16:36.354: JS LOG: [LookingGlass/info] weather@mockturtl#116: Weather Information refreshed
Cinnamon warning: Log level 128: posix_spawn avoided (fd close requested) 

--------------------------------------------------------------------------------------------------------------------------------------
TEMPERATURES:
/sys/devices/pci0000:00/0000:00:02.0/0000:01:00.0/hwmon/hwmon0/temp1_input
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_input
/sys/devices/platform/it87.552/hwmon/hwmon1/temp2_input
/sys/devices/platform/it87.552/hwmon/hwmon1/temp3_input
CRIT:
/sys/devices/pci0000:00/0000:00:02.0/0000:01:00.0/hwmon/hwmon0/temp1_crit
Cjs-Message: 13:04:29.431: JS LOG: [LookingGlass/info] Loaded applet network@cinnamon.org in 224 ms
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cjs-Message: 13:04:29.865: JS LOG: [LookingGlass/info] Loaded applet SpicesUpdate@claudiux in 434 ms
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cjs-Message: 13:04:30.255: JS LOG: Unknown network device type, is 14
Cjs-Message: 13:04:30.718: JS LOG: [LookingGlass/info] Loaded applet menu@cinnamon.org in 854 ms
Cjs-Message: 13:04:30.719: JS LOG: [LookingGlass/info] AppletManager started in 3038 ms
Cjs-Message: 13:04:30.723: JS LOG: [LookingGlass/info] Cinnamon took 3628 ms to start
Cjs-Message: 13:04:31.235: JS LOG: [LookingGlass/info] weather@mockturtl#116: Weather Information refreshed
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) ```

**Comment by Padre2** [comment]
@claudiux   Also checked on Manjaro (Cinnamon 4.4.3) with GPU temperature showing on panel until run `sensors-detect --auto`, then CPU temperature correctly displayed.

**Comment by claudiux** [comment]
@Padre2 Thank you !

It seems that your hardware does not provide any critical temperature (there is no temp*_crit file in your list).

All these `posix_spawn avoided` messages! Grr. Maybe a CSS error. I'm searching.

**Comment by Padre2** [comment]
@claudiux  Not to worry regarding the CPU critical temperature,  I can set my CMOS to give a warning at 60, 70 , 80, or 90 °C, so have no need to depend on the applet for this.  However, as I mentioned at the outset my interest is in how this might be done and perhaps applied to the GPU temperature. My GPU has passive cooling, so I have no fear of a fan failure, but it would be good to have a way to show a warning, if the temperature is too high. For the CPU it presently shows a temperature of 120 °C, where does that value come from, is it generated by the `lm_sensors`?  Regarding the .Xsession-errors warnings, no need to spend your effort for my sake, but I would like to know why they occur, if you do find out.

**Comment by claudiux** [comment]
@Padre2 After reading https://bbs.archlinux.org/viewtopic.php?id=250196, it seems that the files used to obtain the temperatures can change places when Linux starts. For example: `/sys/class/hwmon1` can become`/sys/class/hwmon0`, etc. It is therefore illusory to want to create a definitive list of sensor files. I will try to make sure that they are determined each time the applet is launched.

**Comment by claudiux** [comment]
@Padre2 Can you post here the result of the command `sensors -u`, please?

**Comment by claudiux** [comment]
@Padre2 To try to remove all these warnings of .xsession-errors, can you add in your `~/.local/share/cinnamon/applets/temperature@fevimu` folder this `stylesheet.css` file, please? https://github.com/linuxmint/cinnamon-spices-applets/pull/2755/commits/2d146833d0b0abf7ef0cadb0c626eb9f6bf068c9

Then, reload the code of this applet or restart Cinnamon.

**Comment by Padre2** [comment]
@claudiux  
``sensors -u  (only temperature values listed)`
radeon-pci-0100
Adapter: PCI adapter
temp1:
  temp1_input: 54.500
  temp1_crit: 120.000
  temp1_crit_hyst: 90.000
it8720-isa-0228
Adapter: ISA adapter
  temp1_input: 37.000
  temp1_max: 127.000
  temp1_min: 127.000
  temp1_alarm: 0.000
  temp1_type: 4.000
  temp1_offset: 0.000
  temp1_beep: 1.000
temp2:
  temp2_input: 43.000
  temp2_max: 60.000
  temp2_min: 127.000
  temp2_alarm: 0.000
  temp2_type: 3.000
  temp2_offset: 0.000
  temp2_beep: 1.000
temp3:
  temp3_input: 82.000
  temp3_max: 127.000
  temp3_min: 127.000
  temp3_alarm: 0.000
  temp3_type: 4.000
  temp3_offset: 0.000
  temp3_beep: 1.000``

**Comment by Padre2** [comment]
@claudiux add stylesheet.css to ~/.local/share/cinnamon/applets/temperature@fevimu folder
Does seem to reduce the number of warnings.  If style still in the _js_ file, where is the stylesheet called?
```TEMPERATURES:
/sys/devices/pci0000:00/0000:00:02.0/0000:01:00.0/hwmon/hwmon0/temp1_input
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_input
/sys/devices/platform/it87.552/hwmon/hwmon1/temp2_input
/sys/devices/platform/it87.552/hwmon/hwmon1/temp3_input
CRIT:
/sys/devices/pci0000:00/0000:00:02.0/0000:01:00.0/hwmon/hwmon0/temp1_crit
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cjs-Message: 12:21:44.820: JS LOG: Unknown network device type, is 14
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
blueman-applet 12.21.45 WARNING  PluginManager:147 __load_plugin: Not loading DhcpClient because its conflict has higher priority
Cjs-Message: 12:21:45.607: JS LOG: [LookingGlass/info] Loaded applet menu@cinnamon.org in 1826 ms
Cjs-Message: 12:21:45.607: JS LOG: [LookingGlass/info] AppletManager started in 10418 ms
Cjs-Message: 12:21:45.612: JS LOG: [LookingGlass/info] Cinnamon took 12534 ms to start
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
blueman-applet 12.21.45 WARNING  PluginManager:147 __load_plugin: Not loading PPPSupport because its conflict has higher priority
Cjs-Message: 12:21:46.308: JS LOG: [LookingGlass/info] Adding indicator: blueman
Cjs-Message: 12:21:48.047: JS LOG: [LookingGlass/info] weather@mockturtl#116: Weather Information refreshed
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested)```

**Comment by claudiux** [comment]
@Padre2 Thank you for all. It seems that Cinnamon must be restarted to take into account the new `stylesheet.css` file (which is loaded, if it exists, by the script `/usr/share/cinnamon/js/ui/extension.js` when the applet starts at Cinnamon startup).

**Comment by claudiux** [comment]
@Padre2 The temperature of your CPU seems to be the temp2.

**Comment by Padre2** [comment]
@ claudiux  Yes, agreed and temp3 is the Northbridge (AMD 770) chipset.

**Comment by claudiux** [comment]
@Padre2 I need to know the label of each temp. Can you repost the entire contents of the `sensors -u` return, please?

Bonne fin d'année à vous aussi !

**Comment by Padre2** [comment]
@claudiux Fyi, Cinnamon updated to version 4.4.6 this morning and the temperature displayed on the panel reverted to the GPU temp1. However after restarting it displayed the CPU temp1.!
Also the warnings at the bottom of the .Xsession-errors have now increased to 9.

I have only about 6 months experience with Linux and was not sure of what to do with the script.  I took the following steps:
The script was put into a file and named detect_temp.sh
This file was put into the temperature@fevimu folder
Then in terminal:
```
cd ~/.local/share/cinnamon/applets/temperature@fevimu>
zsh 642 % bash detect_temp.sh
HMAX=2
/sys/class/hwmon/hwmon0/name
radeon
NMAX=1
NCPT=1
/sys/class/hwmon/hwmon1/name
it8720
NMAX=3
NCPT=3
```
Please excuse my ignorance and correct /assist, if I have not understood.

**Comment by claudiux** [comment]
@Padre2 This script has created a file named `sensors_detected.json`, in the same folder. Can you post its contents, please?

**Comment by claudiux** [comment]
@Padre2 Thank you! I'm working hard on this applet. :)

**Comment by Padre2** [comment]
After restarting, the warnings are reducd:
```
Cjs-Message: 12:31:36.975: JS LOG: [LookingGlass/info] Loaded applet temperature@fevimu in 240 ms
Cjs-Message: 12:31:37.292: JS LOG: [LookingGlass/info] Role locked: panellauncher
Cjs-Message: 12:31:37.304: JS LOG: [LookingGlass/info] Loaded applet grouped-window-list@cinnamon.org in 326 ms
Cjs-Message: 12:31:38.294: JS LOG: [LookingGlass/info] Loaded applet weather@mockturtl in 988 ms
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cjs-Message: 12:31:39.674: JS LOG: [LookingGlass/info] Loaded applet network@cinnamon.org in 1378 ms
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cjs-Message: 12:31:41.064: JS LOG: [LookingGlass/info] Loaded applet SpicesUpdate@claudiux in 1390 ms
Cjs-Message: 12:31:42.083: JS LOG: Unknown network device type, is 14
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cjs-Message: 12:31:43.298: JS LOG: [LookingGlass/info] Loaded applet menu@cinnamon.org in 2234 ms
Cjs-Message: 12:31:43.299: JS LOG: [LookingGlass/info] AppletManager started in 11362 ms
Cjs-Message: 12:31:43.304: JS LOG: [LookingGlass/info] Cinnamon took 13192 ms to start
Cjs-Message: 12:31:43.306: JS LOG: [LookingGlass/info] 
TEMPERATURES:
/sys/devices/pci0000:00/0000:00:02.0/0000:01:00.0/hwmon/hwmon0/temp1_input
/sys/devices/platform/it87.552/hwmon/hwmon1/temp1_input
/sys/devices/platform/it87.552/hwmon/hwmon1/temp2_input
/sys/devices/platform/it87.552/hwmon/hwmon1/temp3_input
CRIT:
/sys/devices/pci0000:00/0000:00:02.0/0000:01:00.0/hwmon/hwmon0/temp1_crit
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
blueman-applet 12.31.43 WARNING  PluginManager:147 __load_plugin: Not loading DhcpClient because its conflict has higher priority
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
blueman-applet 12.31.44 WARNING  PluginManager:147 __load_plugin: Not loading PPPSupport because its conflict has higher priority
Cjs-Message: 12:31:45.945: JS LOG: [LookingGlass/info] weather@mockturtl#116: Weather Information refreshed
Cjs-Message: 12:31:45.969: JS LOG: [LookingGlass/info] Adding indicator: blueman
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
```

**Comment by Padre2** [comment]
@claudiux
> I'm working hard on this applet. :)

Shouldn't you be sleeping?  I hadn't finished posting. :-)

**Comment by claudiux** [comment]
@Padre2 Claudiux never sleeps! ;)

Have you gone to see what your computer's BIOS says about the sensors, if any?

**Comment by Padre2** [comment]
@claudiux 
>Have you gone to see what your computer's BIOS says about the sensors, if any?

The Motherboard Manual provides the following:
```I/O Controller  iTE IT8718 chip
Hardware Monitor  System voltage detection
CPU/System temperature detection
CPU/System/Power fan speed detection
CPU overheating warning
CPU/System/Power fan fail warning
CPU/System fan speed control
```
The settings provide for CPU overheating warnings as mentioned earlier.

>Have you take a look at https://www.linuxsecrets.com/archlinux-wiki/wiki.archlinux.org/index.php/Lm_sensors.html ?

Yes, I have read that several times, but may have missed some points.

`zsh 643 % sudo lshw -C system -C processor`                                     
`sudo: lshw: command not found`
Installed LSHW
```
zsh 644 [1] % sudo lshw -C system -C processor
phenom                      
    description: Desktop Computer
    product: GA-MA770-US3
    vendor: Gigabyte Technology Co., Ltd.
    width: 4294967295 bits
    capabilities: smbios-2.4 dmi-2.4 smp vsyscall32
    configuration: boot=normal chassis=desktop uuid=30303146-4430-4444-4632-3138FFFFFFFF
  *-cpu
       description: CPU
       product: AMD Phenom(tm) 9350e Quad-Core Processor
       vendor: Advanced Micro Devices [AMD]
       physical id: 4
       bus info: cpu@0
       version: AMD Phenom(tm) 9350e Quad-Core Processor
       slot: Socket M2
       size: 1GHz
       capacity: 3200MHz
       width: 64 bits
       clock: 200MHz
       capabilities: fpu fpu_exception wp vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp x86-64 3dnowext 3dnow constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid pni monitor cx16 popcnt lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs hw_pstate vmmcall npt lbrv svm_lock cpufreq
```

**Comment by Padre2** [comment]
@claudiux 
>Claudiux never sleeps! ;)

I draw the line at midnight, after that my computer will turn into a pumpkin!

**Comment by claudiux** [comment]
@Padre2 :-)
I suggest you to put this `GA-MA770-US3.conf` file in your `/etc/sensors.d/` folder, before rebooting your computer. I have commented with `#~` the 'Voltages` and `Fans` sections because I'm not completely sure of them. I used the page 52 of the manual of your motherboard (https://download1.gigabyte.com/Files/Manual/motherboard_manual_ga-ma770-us3_e.pdf), the data you gave me above and some config files from https://github.com/lm-sensors/lm-sensors/tree/master/configs/Gigabyte to write it.
```
# libsensors configuration file for Gigabyte GA-MA770-US3 motherboard

chip "it8720-*"

    # lmsensors' values were compared with values from monitoring
    # software running on the same computer under Windows XP SP2
    # abbreviated in comments:
    #   ET6: Gigabyte's EasyTune 6
    #   Everest: Lavalys's Everest Ultimate 4.6.0 (trial version)
    #   OverDrive: AMD's OverDrive 2.1.5

    # Labels are copied from BIOS setup screens when value shown

    #~ # Voltages

    #~ label   in0 "Vcore"
                #~ # BIOS: 1.264V, ET6: 1.260, Everest: 1.26, OverDrive: 1.26
                #~ # No scaling so no compute
                #~ # Values seen using powernow-k8 in "on demand" mode:
                #~ # 1.06, 1.07, 1.25, 1.26, 1.28
                #~ # Nominal values written to /var/log/kern.log by powernow-k8
                #~ # as vid numbers in hex.  For AMD K8, these translate to V by
                #~ # 1.550 - 0.025 * vid. Gave 1.000, 1.175 and 1.200 V
    #~ set     in0_min 1.000 * 0.95  # 95% of nominal min
    #~ set     in0_max 1.200 * 1.05  # 105% of nominal max

    #~ label   in1 "DDR2 1.8V"
                #~ # No scaling so no compute
    #~ set     in1_min 1.8 * 0.95 # 95% of nominal
    #~ set     in1_max 1.8 * 1.05 # 105% of nominal

    #~ label   in2 "+3.3V"
                #~ # No scaling so no compute
    #~ set     in2_min 3.3 * 0.95 # 95% of nominal
    #~ set     in2_max 3.3 * 1.05 # 105% of nominal

    #~ label   in3 "Vcc"
                #~ # BIOS: -, ET6: -, Everest: -, OverDrive: -
                #~ # dmesg reports as VCC
                #~ # Assume standard Winbond scaling resistors
    #~ compute in3 @ * ((6.8/10)+1), @ / ((6.8/10)+1)
    #~ set     in3_min 5 * 0.95 # 95% of nominal
    #~ set     in3_max 5 * 1.05 # 105% of nominal

    #~ label   in4 "5VSB" # Guess
                #~ # BIOS: -, ET6: -, Everest: -, OverDrive (VIN4): 12.74
                #~ # Values seen: 3.15, 3.17, 3.19
                #~ # Assume standard Winbond scaling resistors
    #~ compute in4 @ * ((6.8/10)+1), @ / ((6.8/10)+1)
    #~ set     in4_min 5 * 0.95 # 95% of nominal
    #~ set     in4_max 5 * 1.05 # 105% of nominal

    #~ label   in5 "+12V"
                #~ # BIOS: 12.619V, ET6: 12.610, Everest: -, OverDrive: -
                #~ # Assume standard Winbond scaling resistors
    #~ compute in5 @ * ((30/10)+1), @ / ((30/10)+1)
    #~ set     in5_min 12 * 0.95 # 95% of nominal
    #~ set     in5_max 12 * 1.05 # 105% of nominal

    #~ ignore  in6 # Seen always 4.08, too big (max 4.096) to signify
    #~ ignore  in7 # Seen always 0.03, too small to be significant

    #~ label   in8 "Vbat"
                #~ # BIOS: -, ET6: -, Everest:  3.17, OverDrive: -
                #~ # Connected directly so no compute
    #~ # in8_min and _max not supported by chip

    #~ ignore  cpu0_vid # Not dynamic under lm-sensors 3.0.0

    # Temperatures

    label   temp1  "Current System Temperature"
                # BIOS: 30, ET6: 30, Everest: 33, OverDrive: -
    compute temp1 @ -5, @ +5 # Correct bad sensor on Charles' mobo
    set     temp1_min 0
    set     temp1_max 45 # Tropical ambient max plus a bit

    label   temp2  "Current CPU Temperature"
                # BIOS: 29, ET6: 28, Everest: 32, OverDrive (both cores): 33
    set     temp2_min 0
                # Max must be set to lowest available BIOS "CPU Warning Temperature" - 5. (Adapt it to your case.)
    set     temp2_max 55
                # Crit must be set to lowest available BIOS "CPU Warning Temperature". (Adapt it to your case.)
    set     temp2_crit 60

    ignore  temp3
                # Values seen: 77-79, rising from cold start.
                # Sheilding Northbridge from CPU cooling draft did not change

    #~ # Fans
    #~ # The labels chosen are the ones printed on the motherboard
    #~ label   fan1 "Current CPU FAN Speed"
    #~ set     fan1_min 0 # fancontrol may stop fan
    #~ label   fan2 "Current SYS FAN1 Speed"
    #~ label   fan3 "Current SYS FAN2 Speed"
    #~ ignore  fan4 # "NB_FAN" does not set an RPM signal
    #~ label   fan5 "Current POWER FAN Speed"

    #~ # Beep
    #~ # lm-sensors 3.0.0 does not support on this chip
    #~ # Was enabled by default

chip "k8temp-*"

    # Temperatures
    # For CPU revisions F and G, according to AMD's "Revision
    # Guide for AMD NPT Family 0Fh Processors", "The internal
    # thermal sensor ... is inaccurate". This may be the reason
    # for many reports of k8temp giving absurdly low CPU core
    # temperatures and for there being no generally accepted
    # computation to derive actual temperature from the raw data.
    # The workaround is to ignore core temperatures and use the
    # CPU case temperature as was done before core temperatures
    # were available. On this motherboard the CPU case temperature
    # is reported by the it8718 chip's temp2 (see above).

    # For CPU revisions other than F and G, the proper use of
    # k8temp temperatures may be to compute an average for each
    # core rather than to report two temperatures for each core.
    # For a sample computation see mingus' posting at
    # http://forums.opensuse.org/archives/sls-archives/archives-suse-linux/archives-hardware-support/381563-lm-sensors-2.html#post1793098

    # If you don't trust the readings, you might as well not load
    # the k8temp driver (which, in practice, means blacklisting it,
    # as this driver auto-loads on most systems.
```
After reboot, please run again my `detect_temp.sh` script and post the contents of the resulting `sensors_detected.json` file.

**Comment by Padre2** [comment]
@claudiux 
```
~/.local/share/cinnamon/applets/temperature@fevimu>
zsh 645 % bash detect_temp.sh                                              
HMAX=2
/sys/class/hwmon/hwmon0/name
it8720
NMAX=3
NCPT=3
/sys/class/hwmon/hwmon1/name
radeon
NMAX=1
NCPT=1
```
Contents of the sensors_detected.json file:
```
{
  "it8720": {
    "temp1": {
      "path": "/sys/class/hwmon/hwmon0",
      "input": 36000,
      "max": 127000,
      "type": 4,
      "label": ""
    },
    "temp2": {
      "path": "/sys/class/hwmon/hwmon0",
      "input": 42000,
      "max": 60000,
      "type": 3,
      "label": ""
    },
    "temp3": {
      "path": "/sys/class/hwmon/hwmon0",
      "input": 82000,
      "max": 127000,
      "type": 4,
      "label": ""
    }
  },
  "radeon": {
    "temp1": {
      "path": "/sys/class/hwmon/hwmon1",
      "input": 52500,
      "crit": 120000,
      "label": ""
    }
  }
}
```

**Comment by claudiux** [comment]
@Padre2 Please try `sudo sensors-detect --auto` then re-run the script. Post the `sensors_detected.json` file only if it has changed.

**Comment by Padre2** [comment]
```
{
  "radeon": {
    "temp1": {
      "path": "/sys/class/hwmon/hwmon0",
      "input": 55500,
      "crit": 120000,
      "label": ""
    }
  },
  "it8720": {
    "temp1": {
      "path": "/sys/class/hwmon/hwmon1",
      "input": 39000,
      "max": 127000,
      "type": 4,
      "label": ""
    },
    "temp2": {
      "path": "/sys/class/hwmon/hwmon1",
      "input": 46000,
      "max": 60000,
      "type": 3,
      "label": ""
    },
    "temp3": {
      "path": "/sys/class/hwmon/hwmon1",
      "input": 82000,
      "max": 127000,
      "type": 4,
      "label": ""
    }
  }
}
```

**Comment by claudiux** [comment]
@Padre2 That doesn't work! :(

**Comment by claudiux** [comment]
@Padre2 
Happy new year!
Work is in progress... but not finished.
I just "committed" some improvements into the PR #2755.

**Comment by Padre2** [comment]
Bonne année et bonne santé! Que la force soit avec vous!
Please advise, if any testing needed.

**Comment by Padre2** [comment]
@claudiux  I am not usually loading this applet on Arch, but decided to check it again.  For the record, the temperature displaying in the panel is from the GPU again. I also checked on Ubuntu Cinnamon Remix (Ubuntu 19.10, Cinnamon 4.0.10) and it also is displays the GPU temperature.

**Comment by claudiux** [comment]
Hi @Padre2,
After a too long break, I'm working again on this applet: #3191.

**Comment by Padre2** [comment]
@claudiux  Thanks for the heads up. My Arch system is still going strong with no crashes for over a year and just updated to linux-zen 5.8.1.zen1-1 2 days ago.

**Comment by claudiux** [comment]
@Padre2 I just created the Sensors@claudiux applet. It aims to replace temperature@fevimu.

Can you test it out and give me your feedback, please?

Regards.
Claudiux

**Comment by Padre2** [comment]
@claudiux  Good to hear from you again.
I installed and tested the applet on my Arch Linux version 5.8.10.
The temperature displayed in the tray is the GPU temperature. To be clear about the other temperatures, I attach the screen shot from clicking on the tray temperature and from the xsensors CPU and GPU, which are correct.
![CPU_Temperature_indicator](https://user-images.githubusercontent.com/38849451/94225839-416e3780-ff20-11ea-994f-a30cfea8d01c.png)

![CPU_  MB_temperature_from_xsensors](https://user-images.githubusercontent.com/38849451/94225831-3fa47400-ff20-11ea-9c6f-b40a95937eb4.png)
![Radeon_GPU_temperature_from_xsensors](https://user-images.githubusercontent.com/38849451/94225837-40d5a100-ff20-11ea-8641-7f802eeac50d.png)

**Comment by claudiux** [comment]
@Padre2 Please remove the temperature@fevimu applet and use only the [Sensors@claudiux](https://cinnamon-spices.linuxmint.com/applets/view/337) one. Configure Sensors@claudiux to show in tray the temperature(s) you want.

![image](https://user-images.githubusercontent.com/33965039/94240515-b129f580-ff13-11ea-9709-9a127fab70b0.png)

  1. Check the box: "Display the values of these selected sensors".
  2. Click on the button: "Renew the list of sensors".
  3. Make your selection.
  4. Eventually, click on the button: "Update display".

**Comment by claudiux** [comment]
@Padre2 You also can change the "Shown name". On my computer:

![image](https://user-images.githubusercontent.com/33965039/94243550-d456a400-ff17-11ea-9f6d-733040b0ff53.png)

Result:

![image](https://user-images.githubusercontent.com/33965039/94243896-5ba41780-ff18-11ea-8676-1d3081620f51.png)

**Comment by Padre2** [comment]
@claudiux   Sorry for the misunderstanding.  I did not see the new Sensors Monitor applet and assumed you were still using this old CPU Temperature Indicator applet. I am impressed by the comprehensive scope of the new applet.
I downloaded and installed the Sensors Monitor applet, noticing that it has a '!' yellow warning.  The icon appeared in the panel. I ticked the appropriate boxes, renewed the list of sensors, selected the items and updated the display. 
Perhaps I missed something, but I could not find a way to make the results appear as is shown above.
![Sensors-Monitor](https://user-images.githubusercontent.com/38849451/94258147-9c228600-ff56-11ea-8aff-f8c1cd28fffd.png)
![panel_icon_display](https://user-images.githubusercontent.com/38849451/94258151-9d53b300-ff56-11ea-97bb-59ee783cb4c4.png)

**Comment by claudiux** [comment]
@Padre2 This applet must be configured. Did you have checked the box "Display the values of these selected sensors" of the Temperature tab? Then, clicked on the "Renew the list of sensors" button? Then checked the sensors you want to display in the panel and / or in the tooltip ?

**Comment by Padre2** [comment]
Yes, of course.  Temperatures configured as per attachment.
![Screenshot from 2020-09-25 22-19-19](https://user-images.githubusercontent.com/38849451/94285389-c76c9b80-ff7d-11ea-8791-766c8fe4d2c0.png)

**Comment by claudiux** [comment]
@Padre2  Are there any error messages regarding this applet in your `~/.xsession-errors` file?

**Comment by pnsantos** [comment]
I think I had to click the icon next to the warning sign (the configure button on the manage applets dialog).

If you just add the applet and immediately try to configure it though the taskbar icon (`left click -> General` for example) it won't work (applet won't update). Once it's working though all config changes can be made through the applet itself.

**Comment by Padre2** [comment]
@claudiux  Very busy today and saw nothing in Looking Glass.  I just checked .xsession errors and found the following:
`cinnamon:6296): Cjs-WARNING **: 00:13:55.382: JS ERROR: Exception in callback for signal: sensors-data-available: TypeError: GLib.getenv(...) is null
_formatted_temp@/home/victor/.local/share/cinnamon/applets/Sensors@claudiux/applet.js:739:17
updateUI@/home/victor/.local/share/cinnamon/applets/Sensors@claudiux/applet.js:514:35
anonymous/SensorsApplet/this._connectId<@/home/victor/.local/share/cinnamon/applets/Sensors@claudiux/applet.js:107:75
_emit@resource:///org/cinnamon/cjs/modules/signals.js:142:27
_sensors_reaped@/home/victor/.local/share/cinnamon/applets/Sensors@claudiux/sensorsReaper.js:193:5
anonymous/reap_sensors/<@/home/victor/.local/share/cinnamon/applets/Sensors@claudiux/sensorsReaper.js:97:13
spawnCommandLineAsyncIO/<@/usr/share/cinnamon/js/misc/util.js:233:13`
I think this should help.

**Comment by claudiux** [comment]
@Padre2 
Can you give me the output of the command: `env | egrep "^L[CA]"` please?

@pnsantos 
It is more worrying. I will test that further.

**Comment by claudiux** [comment]
#3235

**Comment by claudiux** [comment]
Sensors@claudiux v1.2.0 is here!

Please try this:
  * Uninstall entirely this applet from System Settings -> Applets (use the [x], not the [-]).
  * Download [this zip file](https://cinnamon-spices.linuxmint.com/files/applets/Sensors@claudiux.zip?203a0837-1286-48b8-a44d-90d0b724efdc) and unzip it in your `~/.local/share/cinnamon/applets` directory. Delete this zip file.
  * Re-open System Settings -> Applets and install Sensors Monitor.

Give me your feedback, please.

**Comment by claudiux** [comment]
#3236

**Comment by Padre2** [comment]
`env | egrep "^L[CA]"`:
`LANG=en_GB.UTF-8~

**Comment by Padre2** [comment]
@claudiux Before proceeding to follow your instructions to uninstall and download the zip file, I noticed the applet update and after installation immediately had temperatures in the panel. Checking further, I noted that the fan speeds are working too. Although are displayed in xsensors, the voltages are not showing and, if selected to display, will prevent tooltips from opening. I did not check the intrusion function.

**Comment by Padre2** [comment]
Have updated to Version 1.2.1 and voltages are displaying in the tooltips. All showing lower than normal, as is also the case in 
![Sensors_Monitors_Version 1 2 1](https://user-images.githubusercontent.com/38849451/94336572-7fe22000-000e-11eb-8c48-ba4587b39497.png)
xsensors.  I will check in Windows.

**Comment by Padre2** [comment]
Here is the display of the temperatures and voltages on this machine running Windows 7. Basically ok, but there are 2 anomalous voltage readings. Old MB, but functioning normally.
![CPUID_HWMonitor_Window_7](https://user-images.githubusercontent.com/38849451/94337254-49f36a80-0013-11eb-8842-ed35e3347b79.PNG)

**Comment by claudiux** [comment]
@Padre2 Maybe copying this conf file in your `/etc/sensors.d` directory, then rebooting your computer? https://github.com/lm-sensors/lm-sensors/blob/master/configs/Gigabyte/MA770-DS3.conf

**Comment by Padre2** [comment]
@claudiux  Thanks for the link.  I downloaded and installed the file, but no significant differences (see attachment).
My motherboard model is  Gigabyte GA-770GMA-USB3 - similar, but perhaps different sensors. I checked for other .conf files but found none. Displaying the voltages are not so important for me and I'm happy with the temperature and fan speed data displays.
![From_MA770-DS3 conf](https://user-images.githubusercontent.com/38849451/94356148-431d3400-00b5-11eb-987b-4c82db40db11.png)

**Comment by Padre2** [comment]
@ claudiux This issue was opened for the CPU Temperature Indicator UUID: temperature@fevimu, which is no longer relevant. May I close it now?

**Comment by claudiux** [comment]
@Padre2 Yes. Thank you for closing it.

**Comment by Padre2** [comment]
@claudiux  Thank you for all the work you did on the temperature sensors and for the new applet.
